src/src/rx_main.cpp

```diff
@@ -1141,28 +1162,68 @@ static void setupSerial()
     CRSF_TX_SERIAL.setRx((PinName)NC);
     CRSF_TX_SERIAL.setTx(GPIO_PIN_RCSIGNAL_TX);
 #endif /* TARGET_RX_GHOST_ATTO_V1 */
-    CRSF_TX_SERIAL.begin(firmwareOptions.uart_baud);
+    CRSF_TX_SERIAL.begin(serialBaud, sbusSerialOutput ? SERIAL_8E2 : SERIAL_8N1);
 #endif /* PLATFORM_STM32 */
+#if defined(TARGET_RX_GHOST_ATTO_V1) || defined(TARGET_RX_FM30_MINI)
+    if (invert)
+    {
+        LL_GPIO_SetPinPull(GPIOA, GPIO_PIN_2, LL_GPIO_PULL_DOWN);
+        USART2->CR1 &= ~USART_CR1_UE;
+        USART2->CR2 |= USART_CR2_TXINV;
+        USART2->CR1 |= USART_CR1_UE;
+    }
+#endif
 
 #if defined(TARGET_RX_FM30_MINI)
     Serial.setRx(GPIO_PIN_DEBUG_RX);
     Serial.setTx(GPIO_PIN_DEBUG_TX);
-    Serial.begin(firmwareOptions.uart_baud); // Same baud as CRSF for simplicity
+    Serial.begin(serialBaud); // Same baud as CRSF for simplicity
 #endif
 
 #if defined(PLATFORM_ESP8266)
-    Serial.begin(firmwareOptions.uart_baud);
-    if (firmwareOptions.invert_tx)
-    {
-        USC0(UART0) |= BIT(UCTXI);
-    }
+    SerialConfig config = sbusSerialOutput ? SERIAL_8E2 : SERIAL_8N1;
+    SerialMode mode = sbusSerialOutput ? SERIAL_TX_ONLY : SERIAL_FULL;
+    Serial.begin(serialBaud, config, mode, -1, invert);
 #elif defined(PLATFORM_ESP32)
-    Serial.begin(firmwareOptions.uart_baud, SERIAL_8N1, -1, -1, firmwareOptions.invert_tx);
+    uint32_t config = sbusSerialOutput ? SERIAL_8E2 : SERIAL_8N1;
+    Serial.begin(serialBaud, config, -1, -1, invert);
 #endif
 
+    if (firmwareOptions.is_airport)
+    {
+        serialIO = new SerialAirPort(CRSF_TX_SERIAL, CRSF_RX_SERIAL);
+    }
+    else if (sbusSerialOutput)
+    {
+        serialIO = new SerialSBUS(CRSF_TX_SERIAL, CRSF_RX_SERIAL);
```

Comment:
Are `CRSF_TX_SERIAL` and `CRSF_RX_SERIAL` really the best names now that we are using those pins for multiple protocols?

Reply:
Probably not. We could rename them `SERIAL_PROTOCOL_TX` and `SERIAL_PROTOCOL_RX`